### PR TITLE
ScreenWithCallback for tabs navigators

### PR DESCRIPTION
**Pull request by Freddy03h**
ScreenWithCallback for BottomTabs, MaterialBottomTabs and MaterialTopTabs navigators, to match the Stack navigator.

### CODE REVIEW

File: src/BottomTabs.re [hunk 1]
```diff
@@ -123,27 +123,50 @@ module Make = (M: {type params;}) => {
...
+  type screenProps('params) = {
+    name: string,
+    options: option(optionsCallback),
+    initialParams: option('params),
+    component:
+      option(
+        React.component({
+          .
+          "navigation": navigation,
+          "route": route,
```
Comment (MoOx) [inline src/BottomTabs.re:157]: Can this be declared as a standard record?

File: src/MaterialBottomTabs.re [hunk 2]
```diff
@@ -63,31 +64,57 @@ module Make = (M: {type params;}) => {
...
+  type screenProps('params) = {
+    name: string,
+    options: option(optionsCallback),
+    initialParams: option('params),
+    component:
+      option(
+        React.component({
+          .
+          "navigation": navigation,
+          "route": route,
```
Comment (MoOx) [inline src/MaterialBottomTabs.re:101]: Same question here (not sure it's doable for some reason)

File: src/BottomTabs.re (hunk 1 above)
Comment (Freddy03h) [reply src/BottomTabs.re:157]: I don't know, I copied the way it was done on the Stack 😅
https://github.com/reason-react-native/reason-react-navigation/blob/a7d40f04e6785ed0c4525ad2b5f63a287282cf7f/src/Stack.re#L127
https://github.com/reason-react-native/reason-react-navigation/blob/a7d40f04e6785ed0c4525ad2b5f63a287282cf7f/src/Stack.re#L269-L282

Comment (MoOx) [reply src/BottomTabs.re:157]: Ok so we could change that later in all places

**Comment by Freddy03h** [comment]
Thank you! 😁

**Pull request merged**